pkg/microservice/aslan/core/workflow/service/workflow/workflow_v4.go

```diff
@@ -938,6 +939,116 @@ func DeleteJiraHookForWorkflowV4(workflowName, hookName string, logger *zap.Suga
+		errMsg := fmt.Sprintf("failed to create jira hook for workflow %s, the error is: %v", workflowName, err)
+		log.Error(errMsg)
+		return e.ErrCreateMeegoHook.AddDesc(errMsg)
+	}
+	return nil
+}
+
+func GetMeegoHookForWorkflowV4Preset(workflowName, hookName string, logger *zap.SugaredLogger) (*commonmodels.MeegoHook, error) {
+	workflow, err := commonrepo.NewWorkflowV4Coll().Find(workflowName)
+	if err != nil {
+		logger.Errorf("Failed to find WorkflowV4: %s, the error is: %v", workflowName, err)
+		return nil, e.ErrGetMeegoHook.AddErr(err)
+	}
+	meegoHook := &commonmodels.MeegoHook{}
+	for _, hook := range workflow.MeegoHookCtls {
+		if hook.Name == hookName {
+			meegoHook = hook
+		}
+	}
+	if err := job.MergeArgs(workflow, meegoHook.WorkflowArg); err != nil {
+		errMsg := fmt.Sprintf("merge workflow args error: %v", err)
+		log.Error(errMsg)
+		return nil, e.ErrGetMeegoHook.AddDesc(errMsg)
+	}
+	meegoHook.WorkflowArg = workflow
+	meegoHook.WorkflowArg.MeegoHookCtls = nil
+	return meegoHook, nil
+}
+
+func ListMeegoHookForWorkflowV4(workflowName string, logger *zap.SugaredLogger) ([]*models.MeegoHook, error) {
+	workflow, err := commonrepo.NewWorkflowV4Coll().Find(workflowName)
+	if err != nil {
+		logger.Errorf("Failed to find WorkflowV4: %s, the error is: %v", workflowName, err)
+		return nil, e.ErrListMeegoHook.AddErr(err)
+	}
+	return workflow.MeegoHookCtls, nil
+}
+
+func UpdateMeegoHookForWorkflowV4(workflowName string, arg *models.MeegoHook, logger *zap.SugaredLogger) error {
+	workflow, err := commonrepo.NewWorkflowV4Coll().Find(workflowName)
+	if err != nil {
+		logger.Errorf("Failed to find WorkflowV4: %s, the error is: %v", workflowName, err)
+		return e.ErrUpdateMeegoHook.AddErr(err)
+	}
+	updated := false
+	for i, hook := range workflow.MeegoHookCtls {
+		if hook.Name == arg.Name {
+			workflow.MeegoHookCtls[i] = arg
+			updated = true
+		}
+	}
+	if !updated {
+		errMsg := fmt.Sprintf("failed to find jira hook %s", arg.Name)
+		log.Error(errMsg)
+		return e.ErrUpdateMeegoHook.AddDesc(errMsg)
+	}
+	if err := commonrepo.NewWorkflowV4Coll().Update(workflow.ID.Hex(), workflow); err != nil {
+		errMsg := fmt.Sprintf("failed to update jira hook for workflow %s, the error is: %v", workflowName, err)
+		log.Error(errMsg)
+		return e.ErrUpdateMeegoHook.AddDesc(errMsg)
+	}
+	return nil
+}
+
+func DeleteMeegoHookForWorkflowV4(workflowName, hookName string, logger *zap.SugaredLogger) error {
+	workflow, err := commonrepo.NewWorkflowV4Coll().Find(workflowName)
+	if err != nil {
+		logger.Errorf("Failed to find WorkflowV4: %s, the error is: %v", workflowName, err)
+		return e.ErrDeleteMeegoHook.AddErr(err)
+	}
+	var list []*models.MeegoHook
+	for _, ctl := range workflow.MeegoHookCtls {
+		if ctl.Name == hookName {
+			continue
+		}
+		list = append(list, ctl)
+	}
+	if len(list) == len(workflow.MeegoHookCtls) {
+		errMsg := fmt.Sprintf("jira hook %s not found", hookName)
```

Comment:
jira => meego

Reply:
Is it possible to delete particular meego hook by using mongo query sql directly?